### PR TITLE
Create Death transformation EOLS DAP

### PR DESCRIPTION
**Pull request by eparkmonster**
## Why
Need to extend DAP End of Life Survey to contain "death" transformations
[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1805)

## This PR
Adds Death transformations to EOLS
Add unit tests
Update eols json schema
Change datatypes for some elements in schema

## Checklist
- [ ] Documentation has been updated as needed.

### CODE REVIEW

File: schema/src/main/jade-fragments/eols_death.fragment.json [hunk 1]
```diff
@@ -15,27 +15,27 @@
         },
         {
             "name": "eol_death_witness_who_you",
-            "datatype": "integer"
+            "datatype": "boolean"
```
Comment (quazi-broad) [inline schema/src/main/jade-fragments/eols_death.fragment.json:18]: looks good, I will double check with Matt about this at our next meeting!

File: dap-etl/transformation/src/main/scala/org/broadinstitute/monster/dap/eols/DeathTransformations.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        if (deathLocation.contains(98))
+          rawRecord.getOptional("eol_death_location_specify")
+        else None,
+      eolDeathWitness = rawRecord.getOptionalNumber("eol_anyone_present"),
+      eolDeathWitnessWhoYou = deathWitness.map(_.contains("1")),
+      eolDeathWitnessWhoFamily = deathWitness.map(_.contains("2")),
+      eolDeathWitnessWhoAcquaintance = deathWitness.map(_.contains("3")),
+      eolDeathWitnessWhoVet = deathWitness.map(_.contains("4")),
+      eolDeathWitnessWhoBoarder = deathWitness.map(_.contains("5")),
+      eolDeathWitnessWhoOther = deathWitness.map(_.contains("98")),
```
Comment (quazi-broad) [inline dap-etl/transformation/src/main/scala/org/broadinstitute/monster/dap/eols/DeathTransformations.scala:23]: The only comment I have is that I have been pulling out the "other" boolean fields as a variable first and then checking that variable when pulling the follow up text field but that's probably overkill for these one off instances so not sure if that should be a real nit

Using RC variable 'eol_who_present' as an example:
```
val deathWitness = rawRecord.fields.get("eol_who_present")
val deathWitnessOther = deathWitness.map(_.contains("98"))
EolsDeath(
    ...
    eolDeathWitnessWhoOther = deathWitnessOther,
    eolDeathWitnessWhoOtherDescription =
        if (deathWitnessOther.contains(true)) rawRecord.getOptional("eol_who_present_specify") else None,
    ...
)
```

**Review by quazi-h — APPROVED** [review]
changes lgtm!

File: dap-etl/transformation/src/main/scala/org/broadinstitute/monster/dap/eols/DeathTransformations.scala (hunk 2 above)
Comment (eparkmonster) [reply dap-etl/transformation/src/main/scala/org/broadinstitute/monster/dap/eols/DeathTransformations.scala:23]: I like that suggestion, makes it consistent and more readable

**Review by quazi-h — APPROVED** [review]

**Review by raaidbroad — APPROVED** [review]

**Pull request merged**